src/infrastructure/figma/figma-service.ts

```diff
@@ -62,6 +64,9 @@ export class FigmaService {
 				},
 				accessToken,
 			);
+
+			parseJsonOfSchemaFromObject(fileResponse, FILE_RESPONSE_SCHEMA);
```

Comment:
**issue:** Please, move the schema validation to `FigmaClient.getFile`:
https://github.com/atlassian-labs/figma-for-jira/blob/1e03d649818d8b57bedb34287afe9ddfa57155d8/src/infrastructure/figma/figma-client/figma-client.ts#L132

It better to encapsulate this responsibility in `FigmaClient` for the following reasons:

- `FigmaClient` guarantees that its methods return data of the specified type, so its consumers should not worry about this concern.
- Schema validation is encapsulated in a single place but not duplicated among consumers.

Please, see the `JiraClient` implementation for an example:
https://github.com/atlassian-labs/figma-for-jira/blob/1e03d649818d8b57bedb34287afe9ddfa57155d8/src/infrastructure/jira/jira-client/jira-client.ts#L110